### PR TITLE
feat: seo sidebar experiment

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Update the `Popular` item to `Explore`.
  - Explore page should look like what is previewed here: https://github.com/dailydotdev/apps/pull/3199
- Remove the `Contribute` section.
- Update the `Manage` to `Activity` section.
  - Add the `Submit a link` (community picks).
  - Move the `Customize` to the `ProfileMenu`.
- On extension:
  - Feeds below should be viewable:
    - Popular.
    - By Upvotes.
    - By Comments.
    - By date.
  - There are certain layouts that we don't support on extension at the moment and we should redirect:
    - Tags
    - Sources
    - Leaderboard
- On mobile:
  - The feed header should swap the `Popular` to `Explore`.
  - There should be a dropdown to list and view the feeds below:
    - Popular.
    - By Upvotes.
    - By Comments.
    - By date.
  - The `Customize` is already part of the gear icon in profile page. 

https://github.com/dailydotdev/apps/assets/13744167/2444802b-f29d-4c8f-b2a6-dc613a791300

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

MI-379 #done

### Preview domain
https://mi-379.preview.app.daily.dev

### CODE REVIEW

File: packages/extension/src/newtab/App.tsx
Comment (sshanzel) [inline packages/extension/src/newtab/App.tsx:94]: Used as a dependency way down.

File: packages/extension/src/newtab/MainFeedPage.tsx
Comment (sshanzel) [inline packages/extension/src/newtab/MainFeedPage.tsx:64]: Used as a dependency way down.

File: packages/shared/src/components/MainFeedLayout.tsx
Comment (sshanzel) [inline packages/shared/src/components/MainFeedLayout.tsx:311]: Had to move as the dropdown goes missing once the feed is empty, making the user unable to switch back.

File: packages/shared/src/hooks/useFeedLayout.ts
Comment (sshanzel) [inline packages/shared/src/hooks/useFeedLayout.ts:109]: Curious to know why we evaluate `isNullOrUndefined` here. It is producing some unwanted behavior.

Result if kept:
Screenshot 2024-06-13 at 1 52 19 PM

File: packages/shared/src/components/ProfileMenu.tsx
Comment (rebelchris) [inline packages/shared/src/components/ProfileMenu.tsx:97]: On the ticket it also mentions Pause new tab, not sure what the final call was?

File: packages/shared/src/components/sidebar/DiscoverSection.tsx
Comment (rebelchris) [inline packages/shared/src/components/sidebar/DiscoverSection.tsx:45]: Where is router here?

File: packages/shared/src/components/sidebar/DiscoverSection.tsx
Comment (rebelchris) [inline packages/shared/src/components/sidebar/DiscoverSection.tsx:121]: Let's add leaderboard already :)

**Review by rebelchris — CHANGES_REQUESTED** [review]
I think we are missing the changes for the horizontal header still.
Otherwise great code, some minor comments

File: packages/shared/src/hooks/useFeedLayout.ts
Comment (nensidosari) [reply packages/shared/src/hooks/useFeedLayout.ts:109]: It's because when server side rendering we don't know if it's laptop or mobile or whatever. I remember this issue, I think there was a bigger issue before, can't remember in detail though, but I think we agreed that we're fine this way since this can only happen if user goes directly to the link which would be SSR, and in mobile usually would go to this page by clicking on the internal links/buttons which would be rendered client side and not have this issue. 

I can try to find the thread where we agreed to this.

File: packages/shared/src/components/ProfileMenu.tsx
Comment (nensidosari) [reply packages/shared/src/components/ProfileMenu.tsx:97]: Final call was to add it to the profile top menu just like in design, can try to find the thread.

File: packages/shared/src/components/sidebar/DiscoverSection.tsx
Comment (nensidosari) [reply packages/shared/src/components/sidebar/DiscoverSection.tsx:121]: that would be /leaderboard right @rebelchris ?

File: packages/shared/src/hooks/useFeedLayout.ts
Comment (nensidosari) [reply packages/shared/src/hooks/useFeedLayout.ts:109]: Took me a while but here it is https://dailydotdev.slack.com/archives/C01L0QXQJNB/p1716278512251279

File: packages/shared/src/components/ProfileMenu.tsx
Comment (nensidosari) [reply packages/shared/src/components/ProfileMenu.tsx:97]: decision here https://dailydotdev.slack.com/archives/C06S6HEUZB9/p1718019097951269?thread_ts=1718017443.993599&cid=C06S6HEUZB9

File: packages/shared/src/hooks/useFeedLayout.ts
Comment (sshanzel) [reply packages/shared/src/hooks/useFeedLayout.ts:109]: After further investigation, it seems the core issue is that the window was not yet ready when the evaluation happened. This is why some checks on load are showing differently from what is expected.

We can see all the issues happening from each previews below.

On this PR, we removed the `isNullOrUndefined` check which product the evaluation to be inaccurate:
https://mi-379.preview.app.daily.dev/bookmarks - the page is supposed to be using some other layout, but since the `isLaptop` is part of the evaluation, it is displaying differently.

On production, it works as intended, but there are some layout shifts happening:
https://app.daily.dev/bookmarks

On this preview, we wait for the window.load event before rendering, no noticeable layout shift and the layout is correct from the get-go:
https://refactor-feed-layout-hook.preview.app.daily.dev/bookmarks

**Comment by nensidosari** [comment]
@sshanzel we also need to change the /explore to /posts based on this https://dailydotdev.slack.com/archives/C06S6HEUZB9/p1718267534996519?thread_ts=1718265658.121159&cid=C06S6HEUZB9. Let's please change here just the url we've added to sidebar/navbar even if it points to 404. I'll create another ticket for changing all urls for it which should also be a blocker for the experiment

File: packages/shared/src/hooks/useFeedLayout.ts
Comment (rebelchris) [reply packages/shared/src/hooks/useFeedLayout.ts:109]: Would it affect other implementations as well?

Comment (sshanzel) [reply packages/shared/src/hooks/useFeedLayout.ts:109]: Yeah, mobile shouldn't be an issue, but on web it happens consistently.

Comment (sshanzel) [reply packages/shared/src/hooks/useFeedLayout.ts:109]: I think I will have to bring back the `isNullOrUndefined` and find a workaround for the issue that I was having. However, the bigger issue still remains.

Comment (nensidosari) [reply packages/shared/src/hooks/useFeedLayout.ts:109]: What's the concrete issue u were having @lee?

> However, the bigger issue still remains.

:hidethepain.....

Comment (capJavert) [reply packages/shared/src/hooks/useFeedLayout.ts:109]: tagged the wrong `@lee` @nensidosari 😆

File: packages/shared/src/components/feeds/FeedNav.tsx
Comment (nensidosari) [inline packages/shared/src/components/feeds/FeedNav.tsx:96]: Let's change this to `posts`

File: packages/shared/src/hooks/useFeedLayout.ts
Comment (sshanzel) [reply packages/shared/src/hooks/useFeedLayout.ts:109]: That the first evaluation can be wrong when checking !isLaptop through the codebase which is the root of most first load layout shift.

Comment (nensidosari) [reply packages/shared/src/hooks/useFeedLayout.ts:109]: Ah yup, known issue now, does it affect anything in particular for this experiment and changes u've done that u think is blocking?

File: packages/shared/src/components/sidebar/DiscoverSection.tsx
Comment (nensidosari) [inline packages/shared/src/components/sidebar/DiscoverSection.tsx:55]: let's change this to /posts

File: packages/shared/src/components/sidebar/DiscoverSection.tsx
Comment (nensidosari) [inline packages/shared/src/components/sidebar/DiscoverSection.tsx:57]: why have this separated here and not on the return below?

File: packages/shared/src/components/sidebar/ManageSection.tsx
Comment (nensidosari) [inline packages/shared/src/components/sidebar/ManageSection.tsx:37]: This should also depend on the sidebar experiement

**Review by nensidosari — CHANGES_REQUESTED** [review]
Few more comments on top of what Chris has already mentioned. Other then that it's looking great :D

File: packages/shared/src/hooks/useFeedLayout.ts
Comment (sshanzel) [reply packages/shared/src/hooks/useFeedLayout.ts:109]: I am getting this when the `is null or undefined` is present.

Screenshot 2024-06-13 at 5 56 14 PM

File: packages/shared/src/components/sidebar/DiscoverSection.tsx
Comment (sshanzel) [reply packages/shared/src/components/sidebar/DiscoverSection.tsx:57]: Because of the change with the order.

File: packages/shared/src/components/sidebar/ManageSection.tsx
Comment (sshanzel) [reply packages/shared/src/components/sidebar/ManageSection.tsx:37]: Yes, I think the evaluations are down below.

File: packages/shared/src/components/sidebar/DiscoverSection.tsx
Comment (sshanzel) [reply packages/shared/src/components/sidebar/DiscoverSection.tsx]: Updated across the codebase 🥳

File: packages/shared/src/hooks/useFeedLayout.ts
Comment (sshanzel) [reply packages/shared/src/hooks/useFeedLayout.ts:109]: Cool. You mean non-blocking for this PR but blocking for the whole experiment?

File: packages/shared/src/components/sidebar/DiscoverSection.tsx
Comment (sshanzel) [reply packages/shared/src/components/sidebar/DiscoverSection.tsx:46]: Ah, was debugging something. Let me remove.

File: packages/shared/src/hooks/useFeedLayout.ts
Comment (nensidosari) [reply packages/shared/src/hooks/useFeedLayout.ts:109]: No I doubt, I'm not actually able to replicate so far on preview, but if it is like the previous ones I mentioned, it only happens if u move to this page via a link, not within the app clicks

File: packages/shared/src/components/ProfileMenu.tsx
Comment (sshanzel) [reply packages/shared/src/components/ProfileMenu.tsx:112]: All of the changes related to Dnd is here: https://github.com/dailydotdev/apps/pull/3215/commits/4824aee9e994ed3a2e3ae4a7285c47ef739582fb

File: packages/shared/src/contexts/DndContext.tsx
Comment (sshanzel) [inline packages/shared/src/contexts/DndContext.tsx:1]: Just moved the file.

File: packages/shared/src/components/feeds/FeedNav.tsx
Comment (rebelchris) [inline packages/shared/src/components/feeds/FeedNav.tsx:111]:
```suggestion
      urls[`${webappUrl}users`] = FeedNavTab.Leaderboard;
```

Endpoint is `/users` for the leaderboard

File: packages/shared/src/components/sidebar/DiscoverSection.tsx
Comment (rebelchris) [reply packages/shared/src/components/sidebar/DiscoverSection.tsx:127]: It's `/users`

**Review by rebelchris — APPROVED** [review]
Cool only the `users` link change I think.
And maybe run Sab through it once :)

File: packages/shared/src/components/ProfileMenu.tsx
Comment (nensidosari) [inline packages/shared/src/components/ProfileMenu.tsx:116]: Think we forgot the onClick here

**Review by nensidosari — COMMENTED** [review]
Few last things and we're good to go 🎉 

Also last thing before we merge this let's also add a message in experiments channel including the data for the experiment like mentioned [here](https://dailydotdev.slack.com/archives/C02JAUF8HJL/p1715175315620999) 🙏 and wait for Ido's approval

File: packages/shared/src/components/ProfileMenu.tsx
Comment (sshanzel) [reply packages/shared/src/components/ProfileMenu.tsx]: Right. Enclosed it within the experiment now 🥳

File: packages/shared/src/components/ProfileMenu.tsx
Comment (sshanzel) [reply packages/shared/src/components/ProfileMenu.tsx:117]: Should now be calling the lazy modal 🚀

**Review by nensidosari — APPROVED** [review]
All good on my side 🎉  

please also do this last thing before we merge 🙏 

> Also last thing before we merge this let's also add a message in experiments channel including the data for the experiment like mentioned [here](https://dailydotdev.slack.com/archives/C02JAUF8HJL/p1715175315620999) 🙏 and wait for Ido's approval

**Comment by sshanzel** [comment]
> All good on my side 🎉
> 
> please also do this last thing before we merge 🙏
> 
> > Also last thing before we merge this let's also add a message in experiments channel including the data for the experiment like mentioned [here](https://dailydotdev.slack.com/archives/C02JAUF8HJL/p1715175315620999) 🙏 and wait for Ido's approval

Yep, I normally send that after merging. Let's also enable this for team members for some time once merged.

**Pull request merged**
